crates/mun_abi/src/lib.rs

```diff
@@ -48,6 +54,54 @@ pub const SET_ALLOCATOR_HANDLE_FN_NAME: &str = "set_allocator_handle";
 #[derive(Clone, Copy, Debug, Eq, PartialEq, Hash, Ord, PartialOrd)]
 pub struct Guid(pub [u8; 16]);
 
+impl From<&[u8]> for Guid {
+    fn from(bytes: &[u8]) -> Self {
+        Guid(md5::compute(&bytes).0)
+    }
+}
+
+#[inline]
+const fn format_hyphenated(src: &[u8; 16]) -> [u8; 36] {
```

Comment:
Does it make sense to add some doc comments here? Maybe move it in the scope of the Display fmt of Guid?